backend/src/apiserver/server/pipeline_server.go

```diff
@@ -592,6 +601,23 @@ func (s *PipelineServer) validatePipelineVersionBeforeCreating(p *model.Pipeline
 	return util.NewInvalidInputError("Pipeline version must have a pipeline spec or a valid source code's URL. PipelineSpec: %s. PipelineSpecURI: %s. CodeSourceUrl: %s. At least one of them must have a valid pipeline spec", p.PipelineSpec, p.PipelineSpecURI, p.CodeSourceUrl)
 }
 
+// Validates a pipeline name to match the requirements. This function matches the constraints in the frontend.
+func validatePipelineName(pipelineName string) error {
+	if pipelineName == "" {
+		return util.NewInvalidInputError("pipeline's name cannot be empty")
+	}
+	if len(pipelineName) > 100 {
```

Comment:
As discussed, this check should be removed.

Reply:
Updated the PR.